tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7966,6 +7966,16 @@ def err_hlsl_unsupported_semantic_index: Error<
   "'%0' is defined with semantic index %1, but only values 0 through %2 are supported">;
 // HLSL Change Ends
 
+// SER Change Starts
+// Shader Execution Reordering
+def err_hlsl_ser_invalid_shader_kind : Error<
+   "Shader kind '%0' incompatible with shader execution reordering (has to be raygeneration, closesthit or miss)">;
+def err_hlsl_reorder_invalid_shader_kind : Error<
+   "Shader kind '%0' incompatible with MaybeReorderThread intrinsic (only in raygeneration)">;
```

Comment:
We should conditionalize this error to reduce how many new warnings we add. Changing the warning text to something like this would allow using a single diagnostic for diagnosing disallowed uses of `HitObject` and `MaybeReorderThread`.

```suggestion
   "%0 is unavailable in shader stage '%1 (requires %2)">;
```

Reply:
Applied the suggestion in [ba11d4f](https://github.com/microsoft/DirectXShaderCompiler/pull/7097/commits/ba11d4f3ffde25c5945852099caf9e983b2deb58)